### DISCUSSION TITLE
Sample of working for .net MAUI blazor

### DISCUSSION THREAD

**Discussion by sohnavcom** [roubachof/MetroLog#38] [Announcements]
Hello do you have a sample as I havent been able to make it work , thanks

**Comment by sohnavcom** [comment]
Thanks for this. I can confirm that going through your sample I have got this to work through a service.

Logs are being written to and shake function is working. But logs are not showing on the screen??

Any ideas ??

**Comment by roubachof** [comment]
have you read this ?

WARNING

you need to add the MemoryTarget if you want to display the logs.
you need to add the StreamingFileTarget if you want to share the logs.

**Comment by sohnavcom** [comment]
`       builder.Logging
.SetMinimumLevel(LogLevel.Trace) // IMPORTANT: set your minimum log level, here Trace
.AddTraceLogger(
    options =>
    {
        options.MinLevel = LogLevel.Trace;
        options.MaxLevel = LogLevel.Critical;
    }) // Will write to the Debug Output
.AddConsoleLogger(
    options =>
    {
        options.MinLevel = LogLevel.Information;
        options.MaxLevel = LogLevel.Critical;
    }) // Will write to the Console Output (logcat for android)
.AddInMemoryLogger(
    options =>
    {
        options.MaxLines = 1024;
       
    })
.AddStreamingFileLogger(
    options =>
    {
        options.RetainDays = 2;
        options.FolderPath = Path.Combine(
            FileSystem.CacheDirectory,
            "MetroLogs");
    });

       builder.Services.AddSingleton(LogOperatorRetriever.Instance);
       builder.Services.AddTransient<MainPage>();`

**Reply by sohnavcom** [reply]
Yes I believe so

**Reply by sohnavcom** [reply]
Hold on..In  your docs the text colour is white. My background in my .net Maui Blazor app is white. You think that maybe it?.

**Reply by roubachof** [reply]
```xml
<?xml version="1.0" encoding="UTF-8"?>

<Application xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="MetroLogSample.Maui.App">
    <Application.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <ResourceDictionary>
                    <Color x:Key="LogPageBackgroundColor">#262626</Color>
                    <Color x:Key="LogPageLogsBackgroundColor">#1D1D1D</Color>
                    <Color x:Key="LogPageLogsBorderColor">#1D1D1D</Color>
                    <Color x:Key="LogPageTextColor">White</Color>
                </ResourceDictionary>
                <ResourceDictionary Source="Resources/Styles/Colors.xaml" />
                <ResourceDictionary Source="Resources/Styles/Styles.xaml" />
            </ResourceDictionary.MergedDictionaries>
        </ResourceDictionary>
    </Application.Resources>
</Application>
```

**Comment by sohnavcom** [comment]
Yeah that was it!! Many thanks for all your help. First Class
